traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/tasks/AuthenticatedRemoteNetworkRequestTask.java

```diff
@@ -39,11 +41,12 @@ public void onError(Error error) {
         } else {
             this.responseHandler = null;
         }
+
+        authTokenFetchTask = new AuthTokenFetchTask(session, context);
```

Comment:
should create the task as part of the execute implementation.

Reason is
a) The work involve in allocating memory etc will be done the background thread
b) The object reference is not held on to for the entirely of this tasks existence

Reply:
updated